sklearn/ensemble/stacking_classifier.py

```diff
@@ -0,0 +1,295 @@
+
+
+        Returns
+        -------
+        self : object
+        """
+        if isinstance(y, np.ndarray) and len(y.shape) > 1 and y.shape[1] > 1:
+            raise NotImplementedError('Multilabel and multi-output'
+                                      ' classification is not supported.')
+
+        if self.estimators is None or len(self.estimators) == 0:
+            raise AttributeError('Invalid `estimators` attribute, `estimators`'
+                                 ' should be a list of (string, estimator)'
+                                 ' tuples')
+
+        if not is_classifier(self.meta_estimator):
+            raise AttributeError('Invalid `meta_estimator` attribute, '
+                                 '`meta_estimator` should be a classifier')
+
+        for name, step in self.estimators:
+            if not hasattr(step, self.method):
+                raise ValueError('Underlying estimator {0} does not support '
+                                 '{1}.'.format(name, self.method))
+
+        for param in kwargs:
+            if not has_fit_parameter(self.meta_estimator, param):
+                raise ValueError('Underlying meta estimator {0} '
+                                 'does not support `{1}`.'.format(
+                                     type(self.meta_estimator), param))
+            for name, step in self.estimators:
+                if not has_fit_parameter(step, param):
+                    raise ValueError('Underlying estimator `{0}` does not '
+                                     'support `{1}`.'.format(name, param))
+
+        self.le_ = LabelEncoder()
+        self.le_.fit(y)
+        self.classes_ = self.le_.classes_
+
+        transformed_y = self.le_.transform(y)
+        if self.cv == 1:  # Do not cross-validation
+            # Parallel fit each estimator
+            prediction_blocks = Parallel(n_jobs=self.n_jobs)(
+                    delayed(_parallel_predict)(clone(clf), self.method,
+                                               X, transformed_y, kwargs)
+                    for _, clf in self.estimators)
+        else:
+            # Use the n_jobs of cross_val_predict
+            prediction_blocks = [(clf.fit(X, y, **kwargs),
+                                  cross_val_predict(clf, X, y, cv=self.cv,
+                                                    method=self.method,
+                                                    fit_params=kwargs,
+                                                    verbose=self.verbose,
+                                                    n_jobs=self.n_jobs))
+                                 for _, clf in self.estimators]
+
+        scores = self._form_meta_inputs([p for _, p in prediction_blocks])
+        self.estimators_ = [est for est, _ in prediction_blocks]
+        self.meta_estimator_ = clone(self.meta_estimator)
+        self.meta_estimator_.fit(scores, transformed_y, **kwargs)
+        return self
+
+    def _form_meta_inputs(self, predicted):
```

Comment:
maybe this doesn't need to be a separate method? maybe just a loop in ``_est_predict`` or something?